### PR TITLE
process.send is async (see: https://github.com/nodejs/node/issues/760)

### PR DESCRIPTION
**Pull request by pieter-v**
using node v4.1.1 a build never finishes, because process.exit on child process is called before the message is delivered to the parent process

### CODE REVIEW

**Comment by dfournier** [comment]
I'll have a look this weekend, thanks for your PR !

**Comment by dfournier** [comment]
Can you please pass the test on your machine? They pass on my computer even with node `v4.0` and `v4.1`

**Comment by pieter-v** [comment]
The original test succeeds, because the response buffer is to small. 
I adapted the test to generate a bigger response buffer. 
Without modifications of run_optimizer, the tests fail (timeout)
I also squashed the commits

Regards

File: run_optimizer.js [hunk 1]
```diff
@@ -22,3 +20,15 @@ process.on('message', function(m) {
...
+  var major = process.versions.node.match(/\d+/)[0];
+  if (major >= 4) {
+    process.send(data, function() {
+      process.exit();
+    });
+  } else {
+    process.send(data);
+    process.exit();
+  }
+}
```
Comment (dfournier) [inline run_optimizer.js]: Please don't delete the last line feed

**Comment by dfournier** [comment]
I don't really like that the node version is checked. Another solution to kill the child could be also to send a message. The message should be send [here](https://github.com/dfournier/broccoli-requirejs/blob/master/index.js#L42) and it should work on 0.x and 4.x version.

Do you agree?

**Comment by pieter-v** [comment]
Looks good, I will try to modify that tomorrow

**Comment by dfournier** [comment]
Fix in #18 

Many thanks @pieter-v for your help ;)

**Pull request closed without merging**